### PR TITLE
Adds vlan_struct

### LINKED ISSUE #2333: Issue: It looks like the new VLAN support doesn't work for Ethernet connections / interfaces
### Describe the issue

I originally opened an issue about having an unsupported installation on HassOS due to Network Manager (https://github.com/home-assistant/operating-system/issues/933) which turned out to be caused by using a VLAN interface which eventually lead to adding support for VLANs to the supervisor here: https://github.com/home-assistant/supervisor/pull/2217.

The supervisor of my install recently got updated to version 2020.12.2 which looks to contain the new VLAN support. I noticed that under Supervisor->System->Host System block that it has an "IP Address" section with a "Change" link next to it but it doesn't show what my IP is (just blank / empty) where as when the test code changes noted in my original opened issue did show my actually IP address of my VLAN interface. Clicking on the "Change" link doesn't seem to do anything and as a side note, the supervisor doesn't say I'm running an unsupported installation like it did before version 2020.12.2

The supervisor logs do report an error (see below) which looks to me the way VLAN support was implemented was only for Wireless interfaces in that the error references my config is missing the "mode" attribute which looks to be related a WiFi NetworkManager option and that the VLAN interface uses wifi_struct.

The issue is in my case my VLAN config is on an Ethernet interface and not a Wireless one. I did try adding:
```
[wifi]
mode=infrastructure
```
to my NetworkManager config file for my VLAN Ethernet interface to see if that would work around he issue but that just prevented NetworkManager from bringing up the interface and thus no connection to the network.

### Steps to reproduce

1. Run a VLAN network configuration 
2. Supervisor gets auto updated to a version that in theory supports VLANs
3. Learn VLAN support doesn't seem to work with Ethernet interfaces

### Enviroment details

```
Version	0.118.4
Installation Type	Home Assistant OS
Development	false
Supervisor	true
Docker	true
Virtual Environment	false
Python Version	3.8.6
Operating System Family	Linux
Operating System Version	5.4.77
CPU Architecture	x86_64
Timezone	America/New_York

Host Operating System	HassOS 4.17
Update Channel	stable
Supervisor Version	2020.12.2
Docker Version	19.03.12
Disk Total	19.0 GB
Disk Used	10.4 GB
Healthy	true
Supported	true
Board	ova
Supervisor API	ok
Version API	ok
```

### Supervisor logs

<details>
<summary>Supervisor logs</summary>

```
20-12-03 14:56:14 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/usr/local/lib/python3.8/site-packages/sentry_sdk/integrations/aiohttp.py", line 123, in sentry_app_handle
    reraise(*_capture_exception(hub))
  File "/usr/local/lib/python3.8/site-packages/sentry_sdk/_compat.py", line 54, in reraise
    raise value
  File "/usr/local/lib/python3.8/site-packages/sentry_sdk/integrations/aiohttp.py", line 113, in sentry_app_handle
    response = await old_handle(self, request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 118, in impl
    return await handler(request)
  File "/usr/src/supervisor/supervisor/api/security.py", line 133, in system_validation
    return await handler(request)
  File "/usr/src/supervisor/supervisor/api/security.py", line 195, in token_validation
    return await handler(request)
  File "/usr/src/supervisor/supervisor/api/utils.py", line 60, in wrap_api
    answer = await method(api, *args, **kwargs)
  File "/usr/src/supervisor/supervisor/api/network.py", line 154, in info
    ATTR_INTERFACES: [
  File "/usr/src/supervisor/supervisor/api/network.py", line 155, in <listcomp>
    interface_struct(interface)
  File "/usr/src/supervisor/supervisor/api/network.py", line 114, in interface_struct
    ATTR_VLAN: wifi_struct(interface.vlan) if interface.vlan else None,
  File "/usr/src/supervisor/supervisor/api/network.py", line 96, in wifi_struct
    ATTR_MODE: config.mode,
AttributeError: 'VlanConfig' object has no attribute 'mode'
```

</details>

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #2333
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/api/network.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def interface_struct(interface: Interface) -> dict:
         ATTR_IPV4: ipconfig_struct(interface.ipv4) if interface.ipv4 else None,
         ATTR_IPV6: ipconfig_struct(interface.ipv6) if interface.ipv6 else None,
         ATTR_WIFI: wifi_struct(interface.wifi) if interface.wifi else None,
-        ATTR_VLAN: wifi_struct(interface.vlan) if interface.vlan else None,
+        ATTR_VLAN: interface.vlan.id if interface.vlan else None,
```
Comment (pvizeli) [inline supervisor/api/network.py:114]: Need to be: `{"id": interface.vlan.id, "parent": interface.vlan.parent}`

Comment (pvizeli) [reply supervisor/api/network.py:114]: https://developers.home-assistant.io/docs/api/supervisor/models#vlan-configuration

Comment (ludeeus) [reply supervisor/api/network.py:114]: `parent` does not exist on `VlanConfig`

```
@attr.s(slots=True)
class VlanConfig:
    """Represent a vlan configuration."""

    id: int = attr.ib()
    interface: str = attr.ib()

```

Comment (pvizeli) [reply supervisor/api/network.py:114]: `{"id": interface.vlan.id, "parent": interface.vlan.interface}`

**Review by pvizeli — APPROVED** [review]

**Pull request merged**